File: du-backend/src/main/java/com/dwin/du/api/entity/Receipt.java
package com.dwin.du.api.entity;
import jakarta.persistence.*;
import lombok.*;
import java.util.Date;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "receipt")
public class Receipt {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "date")
    private Date date;

    @Column(name = "total_price")
    private Double totalPrice;

    @Column(name = "is_settled")
    private boolean settled;
}

File: divvyup/DivvyUp.Shared/Model/Receipt.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace DivvyUp_Shared.Model
{
    public class Receipt
    {
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [ForeignKey("User")]
        [Column("user_id")]
        public int UserId { get; set; }
        public User User { get; set; }

        [Required]
        [Column("name")]
        public string Name { get; set; }

        [Required]
        [Column("date")]
        public DateOnly Date { get; set; }

        [Column("total_price")]
        public decimal TotalPrice { get; set; }

        [Column("is_settled")]
        public bool Settled { get; set; }
    }
}
